Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Worker/Tests/CheckUnifiedApi/SimpleUnfiedAPITest.cs

```diff
@@ -99,12 +80,12 @@ public static double ComputeReduceCube(byte[] inputs)
                    .ToArray();
   }
 
-  public double[] RandomTaskError(double percentageOfFailure = 25)
+  public double[] RandomTaskError(double percentageOfFailure = 0.25)
   {
     var randNum = rd.NextDouble();
     if (randNum < percentageOfFailure / 100)
     {
-      throw new GridServerException("An expected failure in this random call");
+      throw new WorkerApiException("An expected failure in this random call");
```

Comment:
It's not linked to a WorkerApi failure, but a worker side exception. Should change the exception type here